### DISCUSSION TITLE
iOS simulator - how to and problems along the way

### DISCUSSION THREAD

**Discussion by raresn** [airsdk/Adobe-Runtime-Support#2905] [Q&A] [answered]
Hey guys,
I trying to get our app to run inside the iOS simualtor, but failing.
The compile comand is like this:
AdobeAIRSDK/bin/adt -package -target ipa-test-interpreter-simulator -storetype pkcs12 -keystore appstore.p12 -storepass xxx -provisioning-profile appstore.mobileprovision "Agenda-simulator.ipa" "APPLE.xml" "Agenda.swf" "Assets.car" LaunchScreen.storyboardc data icons images res -extdir extensions -platformsdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk

I chose this path because inside the SDKs folder i have iPhoneSimualtor.sdk and iPhoneSimulator16.4.sdk. When using the one without 16.4 we get a warning that we should use something that has x.x at the end.
So this outputs the ipa.

Now, when we try to install and run, we do this
For install:
AdobeAIRSDK/bin/adt -installApp -platform ios -platformsdk /Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk -device ios-simulator -package Agenda-simualtor.ipa
For run:
AdobeAIRSDK/bin/adt -launchApp -platform ios -platformsdk /Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk -device ios-simulator -appid Agenda-simualtor

Both these commands end up with: **Could not locate simctl**
Unfortunately we are stuck here and no idea how to continue as i can't find this exact problem somewhere online.
Here is a screenshot from the xcode locations, maybe it helps assess the situation?
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/1768694/05bf079a-9d9e-41fc-b8d4-7ec78d2bae3c)

@ajwfrost anything i can try here?

**Comment by ajwfrost** [comment, marked as answer]
Hi @raresn - is this path correct? (from your "install" and "run" commands above):
`-platformsdk /Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk`

Earlier (under "compile") you had:
` -platformsdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk`
which makes more sense to me!

Anyway -> ADT tries to find the `simctl` application by going up 5 folders from your SDK path and then adding `/usr/bin/simctl` which would put this at `/Developer/usr/bin/simctl` in your example there, or `/Applications/Xcode.app/Contents/Developer/usr/bin/simctl` if you had the full prefix. So can you please check whether that file exists and is executable?

thanks

**Comment by raresn** [comment]
@ajwfrost  Wow. Sry about this. You are correct - I messed up the paths on the install/compile!
Anyway, this opened a new world for testing for us :)
Just need to figure out how to choose what model of phone or OS we want to Simualtor to have and we are golden!

Thank you again.
